docs/specs/component.md

```diff
@@ -218,6 +168,66 @@ implement since errors are specific to the component.
 *All components* that can drop events MUST emit a `ComponentEventsDropped`
 event in accordance with the [EventsDropped event] requirements.
 
+#### SinkNetworkBytesSent
+
+*Sinks* MUST emit a `SinkNetworkBytesSent` that represents the egress of
+*raw network bytes*.
+
+- Emission
+  - MUST emit immediately after egress of raw network bytes regardless
+    if the transmission was successful or not.
+  - MUST emit *after* processing of the bytes (encryption, compression,
+    filtering, etc.)
+  - MUST NOT emit for pull-based sinks since they do not send data. For
+    example, the `prometheus_exporter` sink MUST NOT emit this event.
```

Comment:
Given the change here to be oriented on network bytes, do we still want to skip pull-based sinks?

Reply:
Nope, good call, updated.